### PR TITLE
feat(config): Add config option for browser socket timeout

### PR DESCRIPTION
**Pull request by ndcornelius**
Sometimes the default timeout for the SocketIO connection attempt isn't enough. There have specifically been issues with the default value of 2000 ms when running with ChromeHeadless. Increasing the timeout value seems to solve this issue in a large number of cases.

This change makes the client timeout value a configuration option. It leaves the default as 2000 ms, but allows for modifying the value if the default is not enough for the browser and environment combination that is targeted.

### CODE REVIEW

**Comment by ndcornelius** [comment]
I signed it!

File: docs/config/01-configuration-file.md [hunk 1]
```diff
@@ -793,6 +793,20 @@ If set then the following fields will be defined and can be overriden:
 All of Karma's urls get prefixed with the `urlRoot`. This is helpful when using proxies, as
 sometimes you might want to proxy a url that is already taken by Karma.
 
+## browserSocketTimeout
+**Type:** Number
+
+**Default:** `2000`
```
Comment (johnjbarton) [inline docs/config/01-configuration-file.md]: The default provided upstream in socket.io is 20000.  Is there any reason we should not use this default as well?  

To me, at worst the extra time will mean that, should the karma server die after launching the browser, the test will last another 18secs. That seems like a small cost for an unusual case.

File: docs/config/01-configuration-file.md [hunk 2]
```diff
@@ -793,6 +793,20 @@ If set then the following fields will be defined and can be overriden:
...
+
+**Description:** Timeout for the client socket connection (in ms).
+
+This configuration represents the amount of time that the client will wait for the socket
+to connect.
+
+When running a browser in different environments, it can take different amounts of time for the
+client socket to connect. If Karma cannot connect within the default timeout, you
```
Comment (johnjbarton) [inline docs/config/01-configuration-file.md]: Please add a sentence explaining what a user will see in the case that the timeout is too small.

File: docs/config/01-configuration-file.md (hunk 1 above)
Comment (johnjbarton) [reply docs/config/01-configuration-file.md]: I changed the default to 20000 in PR #3103 so just change the value here.

Comment (ndcornelius) [reply docs/config/01-configuration-file.md]: Done!

File: docs/config/01-configuration-file.md (hunk 2 above)
Comment (ndcornelius) [reply docs/config/01-configuration-file.md]: Done!

**Pull request merged**